src/interpolation_caches.jl

```diff
@@ -611,3 +611,76 @@ function BSplineApprox(u, t, d, h, pVecType, knotVecType; extrapolate = false)
     c[2:(end - 1)] .= vec(P)
     BSplineApprox(u, t, d, h, p, k, c, pVecType, knotVecType, extrapolate)
 end
+
+"""
+    CubicHermiteInterpolation(du, u, t; extrapolate = false)
```

Comment:
Should we call this `CubicHermiteSpline` instead of `CubicHermiteInterpolation`? (and similarly for quintic)